### PR TITLE
Job Action Added

### PR DESCRIPTION
**Pull request by wahajXgrid**

### CODE REVIEW

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    
+    function_name = "job_restart"
+    finding = Finding(
+        title=f"JOB RESTART",
+        source=FindingSource.MANUAL,
+        aggregation_key=function_name,
+        finding_type=FindingType.REPORT,
+        failure=False,
+    )
+    job_event = event.get_job()
```
Comment (abdullah-shah1) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:25]: Check for None. And all other places

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    """
+    pod = get_job_pod(job_event.metadata.namespace,job_event.metadata.name)
+
+    index = None
+    status_flag = False
+    """
+    Retrieves pod's container information for an OOMKilled pod
+    """
+    for ind,status in enumerate(pod.status.containerStatuses):
+        if status.state.terminated.reason == 'OOMKilled':
```
Comment (abdullah-shah1) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:37]: magicString.

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+def split_num_and_str(num_str:str):
+    num = ''
+    index=None
+    for ind,char in enumerate(num_str):
+       if char.isdigit() or char is '.':
+         num = num+char
+       else:
+         index =ind
+         break
+    return num,num_str[index:]
```
Comment (abdullah-shah1) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:164]: embed style checks in the editor themselves. 
https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    :var max_resource: This variable prevent an infinite loop of job's pod crashing and getting more memory.The action won't increase the memory again when the "Max" limit reached.
+    """
+    increase_to: Optional[float] = 1
+    max_resource: float
+     
+
+@action
+def job_restart_on_oomkilled(event: JobEvent,params: IncreaseResources):
+    
+    function_name = "job_restart"
```
Comment (alikhanxgrid) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:17]: correct function_name

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+    if(split_req > float(split_lim)):
+        split_lim = split_req    
+    
+    return ResourceRequirements(limits={"memory" : (str(split_lim)+lim_unit)},requests={"memory": (str(split_req)+req_unit)})
+    
+
+# Function to get job's Pod
+def get_job_pod(namespace, job):
+    pod_list = PodList.listNamespacedPod(namespace).obj
```
Comment (alikhanxgrid) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:148]: Code might break if the function returns None.

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    :var increase_to: (optional).Users will specify how much they want to increase in each restart.
+    :var max_resource: This variable prevent an infinite loop of job's pod crashing and getting more memory.The action won't increase the memory again when the "Max" limit reached.
+    """
+    increase_to: Optional[float] = 1
+    max_resource: float
+     
+
+@action
+def job_restart_on_oomkilled(event: JobEvent,params: IncreaseResources):
+
```
Comment (alikhanxgrid) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:16]: Add doc string for action.

File: playbooks/robusta_playbooks/job_restart_on_oomkilled.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    increase_to: Optional[float] = 1
+    max_resource: float
+     
+
+@action
+def job_restart_on_oomkilled(event: JobEvent,params: IncreaseResources):
+    
+    function_name = "job_restart"
+    finding = Finding(
+        title=f"JOB RESTART",
```
Comment (alikhanxgrid) [inline playbooks/robusta_playbooks/job_restart_on_oomkilled.py:19]: Rename title to "Job Restart".

**Review by abidiqbalxgrid — APPROVED** [review]

**Review by yousaf-aftab — APPROVED** [review]

**Pull request merged**